lib/auth/result.rb

```diff
@@ -95,6 +97,28 @@ def apply_user_attributes!
     change_made
   end
 
+  def apply_associated_attributes!
+    if authenticator&.provides_groups? && associated_groups.present?
```

Comment:
We should process an empty array, otherwise a user will never be 'removed' from their final group (see spec addition)
```suggestion
    if authenticator&.provides_groups? && !associated_groups.nil?
```